File: PY/Luciano_Chavarria/Algoritmos_de_Busqueda/JumpSearch.py
import math


def jumpSearch(lista, x):
    tamanio = len(lista)
    salto = int(math.floor(math.sqrt(tamanio)))
    previo = int(0)
    while(lista[min(salto, tamanio)-1] < x):
        previo = salto
        salto = salto + int(math.floor(math.sqrt(tamanio)))
        if(previo >= tamanio):
            return -1
    while(lista[previo] < x):
        previo = previo + 1
        if(previo == min(salto, tamanio)):
            return -1
    if(lista[previo] == x):
        return previo
    return -1


if __name__ == '__main__':
    lista = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144]
    numeroabuscar = int(input())
    resultado = jumpSearch(lista, numeroabuscar)
    if(resultado == -1):
        print('No encontrado')
    else:
        print('Encontrado en el indice ', resultado)

File: Java/Bryann Valderrama/Algoritmos de Busqueda/jumpSearch.java
public class Main {

    public static void main(String[] args) {
        int arreglo[] = {0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144};
        int numeroabuscar = 55;
        int resultado = jumpSearch(arreglo, numeroabuscar);
        if (resultado == -1) {
            System.out.println("No encontrado");
        } else {
            System.out.println("Encontrado en el indice " + resultado);
        }
    }

    static int jumpSearch(int arreglo[], int x) {
        int n = arreglo.length;
        int salto = (int) Math.floor(Math.sqrt(n));
        int previo = 0;
        while (arreglo[Math.min(salto, n) - 1] < x) {
            previo = salto;
            salto = salto + (int) Math.floor(Math.sqrt(n));
            if (previo >= n) {
                return -1;
            }
        }
        while (arreglo[previo] < x) {
            previo++;
            if (previo == Math.min(salto, n)) {
                return -1;
            }
        }
        if (arreglo[previo] == x) {
            return previo;
        }
        return -1;
    }
}
